src/corecel/math/EulerRotation.hh

```diff
@@ -0,0 +1,101 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file corecel/math/EulerRotation.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <cmath>
+
+#include "corecel/Constants.hh"
+#include "corecel/Macros.hh"
+#include "geocel/Types.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Perform a vector rotation in eucledian space using Euler angles.
+ *
+ * Constructor takes the three Euler angles \em phi , \em theta , and \em psi ,
+ * which are used for 3 axis rotations, denoted as B, C, and D. The final
+ * general rotation, \f$ A = B \times C \times D \f$, is applied via the
+ * \c operator() .
+ *
+ * The Euler angles follow the x-convention, where:
+ * - \em phi is the angle of the first rotation (D), on the \em z-axis
+ * - \em theta is the angle of the second rotation (C), on the \em x'-axis
+ * (which is the rotated x-axis)
+ * - \em psi is the angle of the third rotation (B), on the \em z'-axis (which
+ * is the rotate z-axis)
+ *
+ * \note For details see https://mathworld.wolfram.com/EulerAngles.html
+ */
+class EulerRotation
```

Comment:
We should remove it if we want to be strict about following rules... but for using it in the near future, my guess it's that it will likely be needed select random directions for the outgoing 3 secondaries produced by the muCF process. Thoughts?

Reply:
If it's not used in this PR, please open a separate one for it. I think something like this already exists in `orange/MatrixUtils.hh` so let's open a discussion separately.

Reply:
Sounds good. Removing until further notice.